src/js/components/Carousel/CarouselChild.js

```diff
@@ -1,70 +1,51 @@
-import React, { useContext, useState, useEffect } from 'react';
+import React, { useState, useEffect, forwardRef } from 'react';
+import {
+  slideLeftCurrent,
+  slideLeftPrevious,
+  slideRightCurrent,
+  slideRightPrevious,
+  StyledCarouselChild,
+} from './StyledCarousel';
 import { CarouselChildPropTypes } from './propTypes';
 
-import { Box } from '../Box';
-import { ThemeContext } from '../../contexts';
-import { defaultProps } from '../../default-props';
+const CarouselChild = forwardRef(
+  ({ children, index, current, previous, direction }, ref) => {
+    const [animation, setAnimation] = useState(undefined);
+    const [display, setDisplay] = useState(
+      current === index ? 'block' : 'none',
+    );
 
-const CarouselChild = ({
-  fill,
-  play,
-  index,
-  activeIndex,
-  priorActiveIndex,
-  children,
-}) => {
-  const theme = useContext(ThemeContext) || defaultProps.theme;
-  const [animation, setAnimation] = useState(undefined);
-  const [visibility, setVisibility] = useState('hidden');
-  useEffect(() => {
-    let timer;
-    if (index === activeIndex) {
-      if (priorActiveIndex !== undefined) {
-        /**
-         * This check will only be false onMount of the component. It ensures
-         * the initial active slide of the Carousel renders with no animation.
-         */
-        setAnimation({
-          type:
-            play || priorActiveIndex < activeIndex ? 'slideLeft' : 'slideRight',
-          size: 'xlarge',
-          duration: theme.carousel.animation.duration,
-        });
+    useEffect(() => {
+      // Previous is only undefined on initialization. This tells the component:
+      // on mount, do not render the first slide with an animation.
+      if (previous === undefined) return;
+      if (index !== current && index !== previous) return;
+
+      if (index === current) {
+        setDisplay('block');
+        if (direction === 'next') setAnimation(slideLeftCurrent);
+        else setAnimation(slideRightCurrent);
+      } else {
+        setTimeout(() => {
```

Comment:
Will this timeout need to be cleared on unmount?

Reply:
Nope, I'll clear it manually good catch.

Reply:
Find a better name for the `animationTimer` variable (maybe)